contracts/governance/extensions/GovernorCountingFractional.sol

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {Governor} from "../Governor.sol";
+import {GovernorCountingSimple} from "./GovernorCountingSimple.sol";
+import {Math} from "../../utils/math/Math.sol";
+
+/**
+ * @notice Extension of {Governor} for 3 option fractional vote counting. When voting, a delegate may split their vote
+ * weight between Against/For/Abstain. This is most useful when the delegate is itself a contract, implementing its
+ * own rules for voting. By allowing a contract-delegate to split its vote weight, the voting preferences of many
+ * disparate token holders can be rolled up into a single vote to the Governor itself. Some example use cases include
+ * voting with tokens that are held by a DeFi pool, voting from L2 with tokens held by a bridge, or voting privately
+ * from a shielded pool using zero knowledge proofs.
+ *
+ * Based on ScopeLift's https://github.com/ScopeLift/flexible-voting/blob/master/src/GovernorCountingFractional.sol
+ */
+abstract contract GovernorCountingFractional is Governor {
+    using Math for *;
+
+    struct ProposalVote {
+        uint256 againstVotes;
+        uint256 forVotes;
+        uint256 abstainVotes;
+        mapping(address voter => uint256) usedVotes;
```

Comment:
I'm confused why this has a different name internally to externally. Honestly I think it's better to have this for consistency, otherwise I need to keep a mental link between the getter and the variable while reading

```suggestion
        mapping(address voter => uint256) voteWeightCast;
```

Reply:
`voteWeightCast` is a public part of the Scopelift module. For compatibility reason, I think we should probably keep the name.

But I hate that name. At the bare minimum it should be `voteWeightCasted`. Lets discuss that on a call.